### PR TITLE
First two columns for a users workshop survey results

### PR DESCRIPTION
**Pull request by mehalshah**

### CODE REVIEW

**Comment by mehalshah** [comment]
todo - permissions

File: dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+require_relative '../../../../../../pegasus/forms/pd_workshop_survey'
```
Comment (aoby) [inline dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb]: That's a lot of `..`s. How about something like `"#{Rails.root}/../pegasus...`

File: dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+class Api::V1::Pd::WorkshopSurveyReportController < Api::V1::Pd::ReportControllerBase
+  load_resource :workshop, class: 'Pd::Workshop'
+
+  FACILITATOR_EFFECTIVENESS_QUESTIONS = [:how_much_learned_s, :how_motivating_s, :how_clearly_presented_s, :how_interesting_s, :how_often_given_feedback_s, :help_quality_s, :how_comfortable_asking_questions_s, :how_often_taught_new_things_s]
+  OVERALL_SUCCESS_QUESTIONS = [:more_prepared_than_before_s, :know_where_to_go_for_help_s, :suitable_for_my_experience_s, :would_recommend_s, :part_of_community_s]
+
```
Comment (aoby) [inline dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb:23]: nit: for readability, one per line:

```
FACILITATOR_EFFECTIVENESS_QUESTIONS = [
  :how_much_learned_s,
  :how_motivating_s,
...
```

File: dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  # GET /api/v1/pd/workshops/:id/aggregate_workshop_score
+  def aggregate_workshop_score
+    survey_report = Hash.new(0)
```
Comment (aoby) [inline dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb]: Any particular reason to set the default hash value to 0?

File: dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  private
+
+  def get_score_for_workshops(workshops)
+    report_column = Hash.new(0)
+    response_count = 0
+
+    workshops.flat_map(&:enrollments).each do |enrollment|
+      response_object = PEGASUS_DB[:forms].where(source_id: enrollment.id).first
+
+      next if response_object.nil?
```
Comment (aoby) [inline dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb:43]: `Forms.source_id` is unique for a given form kind, but not globally (though in practice it is at the moment). To be safe, add `kind: 'PdWorkshopSurvey'` to the where clause, or (preferably) you can instead retrieve the id directly from `enrollment.completed_survey_id`, i.e.

``` ruby
response_object = PEGASUS_DB[:forms].where(id: enrollment.completed_survey_id).first
```

File: dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+      response_count += 1
+
+      survey_response = JSON.parse(response_object[:data])
+
+      survey_response.each do |k, v|
+        k = k.to_sym
```
Comment (aoby) [inline dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb]: could also do `survey_response.symbolize_keys.each do...`

Comment (mehalshah) [reply dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb]: Oh perfect, I was wondering if that was a thing

File: dashboard/app/models/ability.rb [hunk 6]
```diff
@@ -89,7 +89,7 @@ def initialize(user)
         can :manage, Workshop do |workshop|
           workshop.facilitators.include? user
         end
-        can [:read, :start, :end], Pd::Workshop, facilitators: {id: user.id}
+        can [:read, :start, :end, :workshop_survey], Pd::Workshop, facilitators: {id: user.id}
```
Comment (aoby) [inline dashboard/app/models/ability.rb]: This action isn't used anywhere. Did you mean `aggregate_workshop_score`?

Comment (mehalshah) [reply dashboard/app/models/ability.rb]: I did and now I'm confused as to why this worked

Comment (aoby) [reply dashboard/app/models/ability.rb]: Sure it worked just for facilitator? If you're logged in as admin, then you have manage access to `Pd::Workshop` which covers all actions.

This is probably worth adding unit tests for access, like in the [workshops_controller_test](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/controllers/ops/workshops_controller_test.rb)

**Comment by bcjordan** [comment]
@mehals FYI noticed a couple of timeouts during recent builds on this branch in the `rake install` / `rake build` steps, if you update with latest from staging there is a prospective fix in that may fix those.

**Pull request merged**